Source/WebKit/UIProcess/Cocoa/ExtensionCapabilityGranter.mm

```diff
@@ -127,7 +127,15 @@ static bool finalizeGrant(const String& environmentIdentifier, AuxiliaryProcessP
     }
 
     if (grant.isValid()) {
-        iterator->value = WTFMove(grant);
+        auto& existingGrant = iterator->value;
+        ASSERT(!existingGrant.isValid());
+        if (existingGrant.isValid()) {
+            GRANTER_RELEASE_LOG_ERROR(environmentIdentifier, "grant not expected to be valid");
+            granterQueue().dispatch([existingGrant = WTFMove(existingGrant)]() mutable {
```

Comment:
You should call `ExtensionCapabilityGranter::invalidateGrants` here.

Reply:
Ah, good point, addressed in latest patch.

Thanks for reviewing!